igvm/vm.py

```diff
@@ -849,3 +866,94 @@ def vm_performance_value(self) -> float:
         ) * hv_perffactor_src
 
         return float(vm_performance_value)
+
+    def aws_get_instances_overview(self, timeout: int = 5) -> List:
+        """AWS Get Instances Overview
+
+        Load or download the latest instances.json, which contains
+        a complete overview about all instance_types, their configuration,
+        performance and pricing.
+
+        :param: timeout: Timeout value for the head/get request
+
+        :return: VM types overview as list
+        """
+
+        url = AWS_INSTANCES_OVERVIEW_URL
+        file = Path(AWS_INSTANCES_OVERVIEW_FILE)
+        etag_file = Path(AWS_INSTANCES_OVERVIEW_FILE_etag)
+
+        head_req = Request(url, method='HEAD')
+        resp = urlopen(head_req, timeout=timeout)
+        if resp.status == 200:
+            etag = dict(resp.info())['ETag']
+        else:
+            log.debug('Could not retrieve ETag from {}'.format(url))
+
+        if file.exists():
```

Comment:
shortcut: `if file.exists() and etag_file.exists()`. i would also add a `and etag`. this will prevent unnecessary file loading. you can shorten the `if etag and etag == prev_etag:` then accordingly

Reply:
Changed accordingly